csharp/test/Drivers/Apache/Spark/SparkTestEnvironment.cs

```diff
@@ -47,9 +48,11 @@ public override string GetCreateTemporaryTableStatement(string tableName, string
             return string.Format("CREATE TABLE {0} ({1})", tableName, columns);
         }
 
-        public string? GetValueForProtocolVersion(string? hiveValue, string? databrickValue) => ServerType != SparkServerType.Databricks ? hiveValue : databrickValue;
+        public string? GetValueForProtocolVersion(string? hiveValue, string? databrickValue) =>
+            ServerType != SparkServerType.Databricks && ((HiveServer2Connection)Connection).DataTypeConversion.Contains(HiveServer2DataTypeConversion.None) ? hiveValue : databrickValue;
```

Comment:
Use the data type conversion option to help determine the expected type comparison.